src/GridFS/Bucket.php

```diff
@@ -166,6 +175,7 @@ public function __construct(Manager $manager, string $databaseName, array $optio
         $this->databaseName = $databaseName;
         $this->bucketName = $options['bucketName'];
         $this->chunkSizeBytes = $options['chunkSizeBytes'];
+        $this->codec = $options['codec'] ?? null;
```

Comment:
The main problem with the `codec` is that it also applies on incoming data for operations like `insertOne`, while `typeMap` only applies to outgoing data. Applying a codec on the collection broke inserting GridFS files, which is when I noticed that something was up. 

In an ideal world, we'd be handling everything as raw BSON data internally and only decoding it using typeMap or codec when necessary to do so. However, I believe this goes beyond the scope of what we want to do here. I'd be in favour of revisiting this when we make changes to our GridFS API to accommodate the recent request to make the stream protocol available to consumers. The refactoring necessary to accommodate this would also allow us to change the `Bucket` to internally always work with BSON documents and removing this special handling entirely. The collections would then receive a `['root' => 'bson']` type map, and all handling regarding codecs and type maps for the file document would be handled by the `Bucket` class.